crates/template_lib/src/resource/builder/confidential.rs

```diff
@@ -91,48 +188,155 @@ impl ConfidentialResourceBuilder {
         self
     }
 
     /// Sets up the specified `symbol` as the token symbol in the metadata of the resource
+    ///
+    /// # Examples
+    /// ```rust, ignore
+    ///  use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::fungible()
+    ///     .with_token_symbol("MY_TOKEN")
+    ///     .build();
+    /// ```
     pub fn with_token_symbol<S: Into<String>>(mut self, symbol: S) -> Self {
         self.token_symbol = Some(symbol.into());
         self
     }
 
     /// Adds a new metadata entry to the resource
+    ///
+    /// Allows you to add a key-value pair to the resource's metadata.
+    ///
+    /// # Notes
+    ///
+    /// `.add_metadata()` will override any existing metadata with the same key.
+    ///
+    /// # Examples
+    /// ```rust, ignore
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::confidential()
+    ///    .add_metadata("CharacterName", "Tari")
+    ///    .add_metadata("CharacterType", "Mascot")
+    ///    .add_metadata("CharacterLvl", "99")
+    /// .build();
+    /// ```
     pub fn add_metadata<K: Into<String>, V: Into<String>>(mut self, key: K, value: V) -> Self {
         self.metadata.insert(key, value);
         self
     }
 
-    /// Sets up all the metadata entries of the resource
+    /// Replaces the resource's metadata with the given [`Metadata`] object.
+    ///
+    /// Note: This will overwrite any existing metadata, including entries added via `.add_metadata()`.
+    ///
+    /// # Examples
+    /// ```rust, ignore
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// use tari_template_lib::models::Metadata;
+    /// let metadata = Metadata::from([
+    ///     ("Type", "NFT"),
+    ///     ("Creator", "Tari Project"),
+    /// ]);
+    /// let address = ResourceBuilder::confidential()
+    ///     .with_metadata(metadata)
+    ///     .build();
+    /// ```
     pub fn with_metadata(mut self, metadata: Metadata) -> Self {
         self.metadata = metadata;
         self
     }
 
     /// Sets up the image URL of the resource
+    ///
+    /// Allows you to set the image URL of the resource, which can be used in user interfaces to display the token's
+    /// logo or image.
+    ///
+    /// # Examples
+    /// ```rust, ignore
+    /// use tari_template_lib::resource::builder::ResourceBuilder;
+    /// ResourceBuilder::confidential()
+    ///     .with_image_url("https://example.com/my_token_image.png".to_string())
+    ///     .build();
+    /// ```
     pub fn with_image_url(self, url: String) -> Self {
         self.add_metadata(IMAGE_URL, url)
     }
 
-    /// Sets the divisibility of the resource. i.e. the number of decimal places.
-    /// Panic if the divisibility is greater than 18.
+    /// Sets the divisibility of the resource. i.e. the number of decimal places
+    ///
+    /// The default divisibility is 18, which means the smallest unit of the resource is 0.000000000000000001 of the
```

Comment:
I changed this to 8 (apologies for not letting you know) - I'm not too sure what is the most appropriate but 18 seems excessive for confidential (mainly because bulletproofs allows up to u64::MAX units to be stored/transferred confidentially - or just under 18.5 whole units)
```suggestion
    /// The default divisibility is 8, which means the smallest unit of the resource is 0.00000001 of the
```